lib/internal/modules/cjs/loader.js

```diff
@@ -561,14 +559,29 @@ Module._findPath = function(request, paths, isMain) {
     return entry;
 
   let exts;
-  let trailingSlash = request.length > 0 &&
-    StringPrototypeCharCodeAt(request, request.length - 1) ===
-    CHAR_FORWARD_SLASH;
-  if (!trailingSlash) {
-    trailingSlash = RegExpPrototypeExec(trailingSlashRegex, request) !== null;
-  }
+  const trailingSlash = request.length > 0 &&
+    (StringPrototypeCharCodeAt(request, request.length - 1) === CHAR_FORWARD_SLASH || (
```

Comment:
Is there a reason to use charcode instead of:
```suggestion
    (request[request.length - 1] === '/' || (
```
?

Reply:
I believe charcode comparison is faster

Reply:
in recent v8? I’d love to see benchmarks for that!

Reply:
Same here, I'd be up for landing this as is, and open a new PR to run benchmarks and see if that changes something.

Reply:
Totally agree, perf changes should be done separately with benchmarks - i was thinking the simple string comparison was simpler than the charCode, and thus the better choice absent said benchmarks.